src/hooks/VoicePlayer/index.tsx

```diff
@@ -128,62 +129,71 @@ export const VoicePlayerProvider = ({
-      const audioPlayer = new Audio(URL?.createObjectURL?.(audioFile));
-      audioPlayer.id = VOICE_PLAYER_AUDIO_ID;
-      audioPlayer.currentTime = currentAudioUnit.playbackTime;
-      audioPlayer.volume = 1;
-      audioPlayer.loop = false;
-      audioPlayer.onplaying = () => {
-        logger.info('VoicePlayer: OnPlaying event is called from audioPlayer', { groupKey, audioPlayer });
-        voicePlayerDispatcher({
-          type: ON_VOICE_PLAYER_PLAY,
-          payload: { groupKey, audioFile },
-        });
-      };
-      audioPlayer.onpause = () => {
-        logger.info('VoicePlayer: OnPause event is called from audioPlayer', { groupKey, audioPlayer });
-        voicePlayerDispatcher({
-          type: ON_VOICE_PLAYER_PAUSE,
-          payload: { groupKey },
-        });
-      };
-      audioPlayer.ontimeupdate = () => {
+        })
+        .catch(reject);
+    })
+      .then((audioFile: File) => {
+        const voicePlayerRoot = document.getElementById(VOICE_PLAYER_ROOT_ID);
+        logger.info('VoicePlayer: Succeeded getting audio file.', { audioFile });
+        const currentAudioUnit = audioStorage[groupKey] || AudioUnitDefaultValue() as AudioStorageUnit;
+        const audioPlayer = new Audio(URL?.createObjectURL?.(audioFile));
+        audioPlayer.id = VOICE_PLAYER_AUDIO_ID;
+        audioPlayer.currentTime = currentAudioUnit.playbackTime;
+        audioPlayer.volume = 1;
+        audioPlayer.loop = false;
+        audioPlayer.onplaying = () => {
+          logger.info('VoicePlayer: OnPlaying event is called from audioPlayer', { groupKey, audioPlayer });
+          voicePlayerDispatcher({
+            type: ON_VOICE_PLAYER_PLAY,
+            payload: { groupKey, audioFile },
+          });
+        };
+        audioPlayer.onpause = () => {
+          logger.info('VoicePlayer: OnPause event is called from audioPlayer', { groupKey, audioPlayer });
+          voicePlayerDispatcher({
+            type: ON_VOICE_PLAYER_PAUSE,
+            payload: { groupKey },
+          });
+        };
+        audioPlayer.ontimeupdate = () => {
+          voicePlayerDispatcher({
+            type: ON_CURRENT_TIME_UPDATE,
+            payload: { groupKey },
+          });
+        };
+        audioPlayer.dataset.sbGroupId = groupKey;
+        // clean up the previous audio player
+        try {
+          voicePlayerRoot?.childNodes.forEach((node) => {
+            const element = node as HTMLAudioElement;
+            const thisGroupKey = element.dataset?.sbGroupKey;
+            if (thisGroupKey !== groupKey) {
+              element?.pause?.();
+              voicePlayerDispatcher({
+                type: ON_VOICE_PLAYER_PAUSE,
+                payload: { groupKey },
+              });
+              voicePlayerRoot.removeChild(element);
+              logger.info('VoicePlayer: Removed other player.', { element });
+            }
+          });
+        } finally {
+          audioPlayer?.play();
```

Comment:
Is it okay not to set the `onerror` handler or not to use `await audio.play();`? (Handling when audio playback fails)

Reply:
Ah, you are right,
so the original reason is that it was failing to load the audio file, but it will be better to add error handling logic here too

Thank you for reminding me!

Reply:
Now it's twice as safe! LOL